PluginsAndFeatures/azure-toolkit-for-intellij/azure-intellij-plugin-vm/src/main/java/com/microsoft/azure/toolkit/intellij/vm/ssh/UltimateConnectBySshAction.java

```diff
@@ -0,0 +1,115 @@
+
+import javax.annotation.Nonnull;
+
+import java.lang.reflect.InvocationTargetException;
+import java.nio.file.Paths;
+import java.util.Collections;
+import java.util.Objects;
+import java.util.UUID;
+
+/**
+ * connect to selected machine by SSH in Intellij Ultimate.
+ */
+public class UltimateConnectBySshAction {
+    private static final String SSH_CONNECTION_TITLE = "Connect using SSH";
+
+    @AzureOperation(name = "vm.connect_virtual_machine_ssh_ultimate", params = "vm.name()", type = AzureOperation.Type.TASK)
+    public static void connectBySsh(VirtualMachine vm, @Nonnull Project project) {
+        final String configKey = String.format("Azure: %s", vm.getName());
+        final SshConfigConfigurable configurable = new SshConfigConfigurable.Main(project);
+        final SshConfigManager manager = SshConfigManager.getInstance(project);
+        SshConfig existingConfig = manager.findConfigByName(configKey);
+        if (Objects.isNull(existingConfig)) {
+            // add default config
+            final SshConfig defaultConfig = toSshConfig(vm, String.format("Azure: %s", vm.getName()));
+            final SshUiData uiData = new SshUiData(defaultConfig, true);
+            final SshConfigManager.ConfigsData newConfigs = new SshConfigManager.ConfigsData(Collections.emptyList(), Collections.singletonList(uiData));
+            final SshConfigManager.ConfigsData merged = manager.getLastSavedAndCurrentData().createMerged(newConfigs);
+            manager.applyData(merged.getCurrentData(), new SshConfigManager.Listener() {
+                @Override
+                public void sshConfigsChanged() {
+                    SshConfigManager.Listener.super.sshConfigsChanged();
+                }
+            });
+            existingConfig = defaultConfig;
+        }
+        openPredefinedTerminalSshSessionAction(existingConfig, project);
+    }
+
+    private static void openPredefinedTerminalSshSessionAction(SshConfig existingConfig, Project project) {
+        final SshConsoleOptionsProvider provider = SshConsoleOptionsProvider.getInstance(project);
+        final RemoteCredentials sshCredential = new SshUiData(existingConfig, true);
+        final SshTerminalCachingRunner runner = new SshTerminalCachingRunner(project, sshCredential, provider.getCharset());
+        AzureTaskManager.getInstance().runInBackground(SSH_CONNECTION_TITLE,() ->
+                connectToSshUnderProgress(project, runner, sshCredential)
+        );
+    }
+
+    private static void connectToSshUnderProgress(final @NotNull Project project, final @NotNull SshTerminalCachingRunner runner, final @NotNull RemoteCredentials data) {
+        try {
+            runner.connect();
+            AppUIUtil.invokeLaterIfProjectAlive(project, () ->
+                    TerminalView.getInstance(project).createNewSession(runner));
+        } catch (final RemoteSdkException e) {
+            AppUIUtil.invokeLaterIfProjectAlive(project, () ->
```

Comment:
we may user `AzureMessager` to show notification and provide show config window action in the notifiaction

Reply:
done